scalatest-test/src/test/scala/org/scalatest/prop/GeneratorSpec.scala

```diff
@@ -1722,6 +1722,34 @@ class GeneratorSpec extends FunSpec with Matchers {
         edges should contain (NonZeroInt(1))
         edges should contain (NonZeroInt.MinValue)
       }
+      it("should produce NonZeroInt canonical values") {
+        import Generator._
+        val gen = nonZeroIntGenerator
+        val (canonicals, _) = gen.canonicals(Randomizer.default)
+        canonicals.toList shouldBe List(NonZeroInt(1), NonZeroInt(-1), NonZeroInt(2), NonZeroInt(-2), NonZeroInt(3), NonZeroInt(-3))
+      }
+      it("should shrink NonZeroInts by repeatedly halving and negating") {
```

Comment:
Is it appropriate to specify this quite so tightly?  This feels like testing the implementation rather than the requirement.  Again not end of the world, but it feels too coupled to the implementation...

Reply:
Not sure. I think that's the spec in this case. Let's leave this one as is for now.